### PR TITLE
schemas: backward compatibility check of schemas

### LINKED ISSUE #1076: api: check schemas backward compatibility using inveniosoftware/doschema.
We need to use the validation function in inveniosoftware/doschema for the validation of our block and community schemas.

The validation should be done in the programmatic API. The REST API should have an error handler which forwards the exception as a REST Exception [as it is done here](https://github.com/EUDAT-B2SHARE/b2share/blob/def0d0cce6d0677bab5c4354c44a7b81f478e467/b2share/modules/records/errors.py#L49).

Note: DoSchema is not released yet so the travis tests will work only in devel mode:
- Add `-e git+git://github.com/inveniosoftware/doschema.git#egg=doschema` in this file https://github.com/EUDAT-B2SHARE/b2share/blob/evolution/requirements-devel.txt
- Add doschema to setup.py.

The corresponding PR will of course be merged only once doschema is released so that the branch is not broken.

### PR DESCRIPTION
**Pull request by nharraud**
* NEW Checks backward compatibility of JSON Schemas. (based on #1096)
  (closes #1076)

Signed-off-by: Nicolas Harraudeau <nicolas.harraudeau@cern.ch>
Co-authored-by: PaulinaLach <paulina.malgorzata.lach@cern.ch>

### CODE REVIEW

File: b2share/modules/schemas/errors.py [hunk 1]
```diff
@@ -25,38 +25,45 @@
...
-class B2ShareSchemasError(Exception):
+
+class B2ShareSchemasError(RESTException):
     """Base class for the B2Share Schemas module."""
-    pass
 
+    def __init__(self, description=None, code=None):
+        """Constructor"""
+        super(B2ShareSchemasError, self).__init__()
+        self.code = code if code is not None else 500
```
Comment (emanueldima) [inline b2share/modules/schemas/errors.py]: This sets the default `code` value to 500 in all the subclasses' instances, which I believe is not the intended behaviour. Do you mean:

````
class B2ShareSchemasError(RESTException):
    code = 500
    def __init__(self, description=None, code=None):
        ....
        if code:
            self.code = code
````

File: b2share/modules/schemas/errors.py [hunk 2]
```diff
@@ -25,38 +25,45 @@
...
 
 #
 # JSON SCHEMA ERRORS
 #
 
 class InvalidJSONSchemaError(B2ShareSchemasError):
     """Exception raised when a JSON Schema is considered as invalid."""
-    pass
-
+    code = 400
```
Comment (emanueldima) [inline b2share/modules/schemas/errors.py]: Related to previous comment.

File: b2share/modules/schemas/errors.py (hunk 1 above)
Comment (nharraud) [inline b2share/modules/schemas/errors.py]: Thanks for catching that. I'll improve it.

**Comment by nharraud** [comment]
I will wait for PR 1384 to be merged before and rerun the tests, just in case.

**Pull request merged**